### PR TITLE
[mergebot] Fix mergebot allow revert of codev diff 

### LINKED ISSUE #1381: Revert bot should not allow github reverts of internally landed (codev changes). Revert bot should already have this logic, but we have a bug somewhere
When an codev (internal) PR gets pushed externally, the revert for that PR should happen externally too.

If a dev tries to revert a codev PR, revertbot should reply with an error message explaining that the dev should click the Diff link from the PR body to go to the internal diff page and click the "Unland" button there to revert the PR.

Not doing so risks the FB internal and external version of the code going out of sync

Sample PR: https://github.com/pytorch/pytorch/pull/91340

### PR DESCRIPTION
**Pull request by clee2000**
mergebot was allowing non facebook-github-bot users to revert codev diffs when it shouldnt be allowed

Fixes https://github.com/pytorch/test-infra/issues/1381

### CODE REVIEW

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -1257,7 +1257,7 @@ def validate_revert(repo: GitRepo, pr: GitHubPR, *,
         commit_sha = commits[0]
     msg = repo.commit_message(commit_sha)
     rc = RE_DIFF_REV.search(msg)
-    if rc is not None and not can_skip_internal_checks:
+    if rc is not None and not skip_internal_checks:
         raise PostCommentError(f"Can't revert PR that was landed via phabricator as {rc.group(1)}")
```
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:1261]: Nice! Can you please also update the comment error to explain the correct way to revert the PR? (Click the diff link in the PR body to go to the diff page and click the "unland" button on that page)

**Review by ZainRizvi — APPROVED** [review]

**Review by kit1980 — APPROVED** [review]

**Review by malfet — APPROVED** [review]
🙈 (I.e. that was a fantastic mistaken on my end)

Can we add a test? :P

**Pull request closed without merging**